<launch>
    <arg name="cam_id" default="1"/>
    <arg name="cam_type" default="usb_cam"/>
    <arg name="device_id" default="1"/>
    
    <group ns="$(eval 'cam_' + str(arg('cam_id')))" if="$(eval arg('cam_type') == 'kinect')">
        <arg name="kinect_device_id" value="$(eval '#' + str(arg('device_id')))"/>
        
        <include file="$(find freenect_launch)/launch/freenect.launch">
            <arg name="device_id" value="$(arg kinect_device_id)"/>
        </include>
        
        <group ns = "camera">
            <include file="/home/ur3/multi-camera-ros/catkin_ws_camera/launch/marker_publisher.launch">
                <arg name="cam_id" value="$(eval 'cam_' + str(arg('cam_id')))"/>
            </include>
        </group>    
    
    </group>

    <group ns="$(eval 'cam_' + str(arg('cam_id')))" if="$(eval arg('cam_type') == 'usb_cam')">
        <arg name="usb_cam_device_id" value="$(eval arg('device_id') - 1)"/>
        <group ns="camera">
            
            <node name="rgb" pkg="usb_cam" type="usb_cam_node" output="screen" >
                <param name="video_device" value="/dev/video$(arg usb_cam_device_id)" />
                <param name="image_width" value="640" />
                <param name="image_height" value="480" />
                <param name="pixel_format" value="yuyv" />
                <param name="framerate" value="30" />
                <param name="camera_frame_id" value="$(eval 'cam_' + str(arg('cam_id')) + ' usb_cam')" />
                <param name="io_method" value="mmap"/>
            </node>
            
            <include file="/home/ur3/multi-camera-ros/catkin_ws_camera/launch/marker_publisher.launch">
                <arg name="cam_id" value="$(eval 'cam_' + str(arg('cam_id')))"/>
            </include>
        </group>
    </group>


</launch>